src/searchModal/advancedSearch.js

```diff
@@ -0,0 +1,445 @@
+            $advancedCriteriaContainer.removeClass('scroll-separator-top');
+            $advancedCriteriaContainer.removeClass('scroll-separator-bottom');
+            $advancedCriteriaContainer.empty();
+            _.forEach(criteriaState, criterion => {
+                criterion.rendered = false;
+                criterion.value = undefined;
+            });
+        },
+        /**
+         * Builds substring of search query with the advanced criteria conditions
+         */
+        getAdvancedCriteriaQuery: function () {
+            const advancedSearchCriteria = _.filter(criteriaState, criterion => criterion.rendered === true);
+            let query = '';
+
+            advancedSearchCriteria.forEach(renderedCriterion => {
+                if (renderedCriterion.type === 'text') {
+                    if (renderedCriterion.value && renderedCriterion.value.trim() !== '') {
+                        query += ` AND ${renderedCriterion.label}:${renderedCriterion.value.trim()}`;
+                    }
+                } else if (renderedCriterion.type === 'list') {
+                    if (renderedCriterion.value && renderedCriterion.value.length > 0) {
+                        query += ` AND ${renderedCriterion.label}:${renderedCriterion.value.join(' OR ')}`;
+                    }
+                }
+            });
+
+            return query;
+        }
+    })
+        .setTemplate(advancedSearchTpl)
+        .on('render', () => {
+            initUiSelectors();
+            initAddCriteriaSelector();
+            initcriteriaState();
+        });
+
+    /**
+     * Inits template selectors and scroll animation
+     */
+    function initUiSelectors() {
+        $container = instance.getElement();
+        $addCriteriaInput = $('.add-criteria-container a', $container);
+        $criteriaSelect = $('.add-criteria-container select', $container);
+        $advancedCriteriaContainer = $('.advanced-criteria-container', $container);
+
+        $advancedCriteriaContainer.on('scroll', animateScroll);
+    }
+
+    /**
+     * Inits select2 on criteria select and its UX logic
+     */
+    function initAddCriteriaSelector() {
+        $criteriaSelect.select2({
+            containerCssClass: 'criteria-select2',
+            dropdownCssClass: 'criteria-dropdown-select2',
+            sortResults: results => _.sortBy(results, ['text'])
+        });
+
+        // open dropdown when user clicks on add criteria input
+        $addCriteriaInput.on('click', () => {
+            $criteriaSelect.select2('open');
+            // if dropdown is opened above addCriteria input, top property is slightly decreased to avoid overlapping with addCriteria icon
+            if ($('.criteria-dropdown-select2').hasClass('select2-drop-above')) {
+                $('.criteria-dropdown-select2').css(
+                    'top',
+                    $('.criteria-dropdown-select2').css('top').split('px')[0] - 10 + 'px'
+                );
```

Comment:
I don't see how this could be done with pure css.

The purpose of this declaration is to fix a small visual glitch when criteria list is opened and is being rendered above _add criteria_ button, causing the list of options to overlay the (+) icon. The `top` property of `.criteria-dropdown-select2` is not fixed, instead it depends of  where is the _add criteria_ button rendered, that is why I need to get access to it and decrease it 10px. Ideally I would modify the logic that manages if the dropdown must be rendered up or above the _add criteria_ button, and decrease the 10px there, but this is managed internally by `select2` so I don't have access to it.

Hope this gif helps to understand: 

![pr-0](https://user-images.githubusercontent.com/14041944/95877467-6cf07f00-0d74-11eb-98bc-e5f444227c0b.gif)